### PR TITLE
Add Alpine package provider

### PR DESCRIPTION
**Pull request by damacus**
Adds Alpine package provider.

- [x] Add Alpine tests
- [x] Stub apk grep command for alpine
- [x] Resolve (disable for now) rubocop ABC/CyclomaticComplexity/PerceivedComplexity

### CODE REVIEW

File: lib/resources/package.rb [hunk 1]
```diff
@@ -44,6 +44,8 @@ def initialize(package_name, opts = {}) # rubocop:disable Metrics/AbcSize
         @pkgman = SolarisPkg.new(inspec)
       elsif ['hpux'].include?(os[:family])
         @pkgman = HpuxPkg.new(inspec)
+      elsif os.alpine?
```
Comment (chris-rock) [inline lib/resources/package.rb]: I would use `['alpine'].include?(os[:name])`, we do not need a new helper for alpine

File: docs/resources/os.md.erb [hunk 2]
```diff
@@ -70,6 +70,7 @@ The `os` audit resource includes a collection of helpers that enable more granul
 * `suse?`
 * `unix?`
 * `windows?`
+* `alpine?`
```
Comment (chris-rock) [inline docs/resources/os.md.erb]: I think we do not need alpine here

File: test/unit/resources/package_test.rb [hunk 3]
```diff
@@ -152,6 +152,15 @@
...
 
+  # alpine
+  it 'verify Alpine package processing' do
+    resource = MockLoader.new(:alpine).load_resource('package', 'git')
+    pkg = { name: 'git', installed: true, version: '2.15.0-r1', type: 'pkg' }
+    _(resource.installed?).must_equal true
+    _(resource.version).must_equal '2.15.0-r1'
+    _(resource.info).must_equal pkg
+  end
+
```
Comment (chris-rock) [inline test/unit/resources/package_test.rb:163]: Great work, you need to add the command output for `apk info -vv --no-network | grep git` in https://github.com/inspec/inspec/tree/master/test/unit/mock/cmd and the command registration similar to https://github.com/inspec/inspec/blob/master/test/helper.rb#L244

**Review by chris-rock — DISMISSED** [review]
Great work @damacus I added some comments

File: test/unit/resources/package_test.rb (hunk 3 above)
Comment (damacus) [reply test/unit/resources/package_test.rb:163]: Thank you, I was wondering how/where to stub and mock that data.

**Comment by damacus** [comment]
Looks like the functional test suite is failing 🤔

**Comment by damacus** [comment]
@chris-rock I think this is ready to merge now :+1:

**Review by jquick — APPROVED** [review]
Looks good! Thanks for this @damacus

**Review by jerryaldrichiii — APPROVED** [review]
Great work @damacus. Super valuable for the community!

File: lib/resources/package.rb [hunk 4]
```diff
@@ -19,8 +19,10 @@ class Package < Inspec.resource(1)
         its('version') { should eq 1.9.5 }
       end
     "
-
-    def initialize(package_name, opts = {}) # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
+    def initialize(package_name, opts = {})
```
Comment (jquick) [inline lib/resources/package.rb:25]: In the future we usually just cascade the rubocop disables on the method call. That way we don't have to enable after. Example:
```
def initialize(package_name, opts = {}) # rubocop:disable Metrics/AbcSize Metrics/CyclomaticComplexity  Metrics/PerceivedComplexity
...
end
```

**Comment by damacus** [comment]
Awesome thanks for letting me know about that rubocop rule. I might have a
think about how to refactor that method in the future

**Comment by miah** [comment]
Thanks for the contribution @damacus =)

**Pull request merged**